linter_plugin.py

```diff
@@ -1,31 +1,50 @@
-"""Certbot ACME PyLint plugin.
-
+"""Certbot PyLint plugin.
 http://docs.pylint.org/plugins.html
-
 """
-from astroid import MANAGER
-from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers.utils import check_messages
+
+
+# Module in theses packages can import the os module.
+WHITELIST_PACKAGES = ['acme', 'certbot_compatibility_test', 'letshelp_certbot', 'lock_test']
 
 
-def register(unused_linter):
-    """Register this module as PyLint plugin."""
-    pass
+class ForbidStandardOsModule(BaseChecker):
+    """
+    This checker ensures that standard os module is not imported by certbot modules.
+    Otherwise a 'os-module-forbidden' error will be registered for the faulty lines.
+    """
+    __implements__ = IAstroidChecker
 
+    name = 'forbid-os-module'
+    msgs = {
+        'E5001': (
+            'Forbidden use of os module, certbot.compat.os must be used instead',
+            'os-module-forbidden',
+            'Some methods from standard os modules cannot be used for security reasons on Windows: '
+            'the safe wrapper certbot.compat.os must be used instead in Certbot.'
+        )
+    }
+    priority = -1
 
-def _transform(cls):
-    # fix the "no-member" error on instances of
-    # letsencrypt.acme.util.ImmutableMap subclasses (instance
-    # attributes are initialized dynamically based on __slots__)
+    @check_messages('os-module-forbidden')
+    def visit_import(self, node):
+        if 'os' in [name[0] for name in node.names] and not _check_disabled(node):
+            self.add_message('os-module-forbidden', node=node)
 
-    # TODO: this is too broad and applies to any tested class...
+    @check_messages('os-module-forbidden')
+    def visit_importfrom(self, node):
+        if node.modname == 'os' and not _check_disabled(node):
+            self.add_message('os-module-forbidden', node=node)
 
-    #if cls.slots() is not None:
-    #    for slot in cls.slots():
-    #        cls.locals[slot.value] = [nodes.EmptyNode()]
 
-    if cls.name == 'JSONObjectWithFields':
-        # _fields is magically introduced by JSONObjectWithFieldsMeta
-        cls.locals['_fields'] = [nodes.EmptyNode()]
+def register(linter):
+    """Register this module as a Pylint plugin."""
```

Comment:
nit: We're not registering this module as a plugin are we?

Reply:
No indeed. It is hook. Updated the documentation accordingly.